libsolidity/analysis/TypeChecker.cpp

```diff
@@ -532,9 +551,33 @@ bool TypeChecker::visit(VariableDeclaration const& _variable)
 		{
 			solAssert(!result.message().empty(), "Expected detailed error message");
 			m_errorReporter.typeError(1534_error, _variable.location(), result.message());
+			return false;
 		}
 	}
 
+	if (varType->dataStoredIn(DataLocation::Storage))
+	{
+		if (storageSizeUpperBound(*_variable.type()) >= bigint(1) << 64)
+			m_errorReporter.warning(
+				3408_error,
+				_variable.location(),
+				"Variable " + util::escapeAndQuoteString(_variable.name()) +
+				" covers a large part of storage and thus makes collisions likely. "
+				"Either use mappings or dynamic arrays and allow their size to be increased only "
+				"in small quantities per transaction."
+			);
```

Comment:
If we're feeling perfectionistic, we could change the error message depending on the case...
If it's a state variable we could say ``Variable ... covers a large part...`` and if it isn't, we could say ``The type ... covers`` using the type of the variable - but I wouldn't even have noticed, if you had not had the unnamed variable with the ``Variable "" covers...`` message in the tests, so maybe we really don't have to account for this and it's just fine :-).

Reply:
@ekpyron 
> If we're feeling perfectionistic, we could change the error message depending on the case...

Done, I also used `_variable.typeName()->location()` instead of `_variable.location()` when referring to types and subtypes, made a couple of minor changes and squashed.